documentation/specs/src/economics/public-goods-funding.md

```diff
@@ -38,8 +28,237 @@ Namada groups public goods into four categories, with earmarked pools of funding
+
+- At epoch 49, Alice submits the vote:
+
+```rust
+struct OnChainVote {
+    id: 2,
+    voter: 0xalice,
+    yay: proposalVote,
+}
+```
+
+Whereby the `proposalVote` includes
+```rust
+HashSet<(address: 0xBobCharlieMultisig, spending_cap: 1.0)>
+```
+
+- At epoch 49, Bob and Charlie submit identical transactions. Charlie also 
+
+- At epoch 50, Dave votes `Nay` on the proposal.
+
+- At epoch 51, Elsa votes `Yay` but on the multisigs `0xElsaAndSisterMultisig` AND `0xBobCharlieMultisig`.
+
+- At epoch 54, the voting period ends and the votes are tallied. Since 80% > 33% of the voting power voted `Yay` on this proposal, a new council will be elected. The council that received the most votes, in this case `0xBobCharlieMultisig` is elected the new PGF council.
+
+- At epoch 57, Bob and Charlie have the effective power to carry out Public Goods Funding transactions.
+````
+
+### End of Term Summary
+
+At the end of each term, the council is encouraged to submit a "summary"  which describes the funding decisions the councils have made and their reasoning for these decisions. This summary will act as an assessment of the council and will be the information point for governance to decide whether to re-elect the council.
+
+## Mechanism
+
+Once elected and instantiated, members of the PGF council will then unilaterally be able to propose and sign transactions for this purpose. The PGF council multisig will have an "allowance" to spend up to the `PGF` internal address's balance multiplied by the `spending_cap` variable.  Consensus on these transactions, in addition to motivation behind them will be handled off-chain, and should be recorded for the purposes of the "End of Term Summary".
+
+
+### PGF council transactions
+The PGF council members will be responsible for collecting signatures offline. One member will then be responsinble for submitting a transaction containing at least $k $ out of the signatures.
+
+The collecting member of the council will then be responsible for submitting this tx through the multisig. The multisig will only accept the tx if this is true.
+
+Note that there is no cap to $n$ in $k$-out-of-$n$ apart from the limitations of the multisig account.
+
+The PGF council should be able to make both retroactive and continuous public funding transactions. Retroactive public funding transactions should be straightforward and implement no additional logic to a normal transfer.
+
+However, for continuous PGF (cPGF), the council should be able to submit a one time transaction which indicates the recipient addresses that should be eligble for receiveing cPGF. 
+
+The following data is attached to the PGF transaction and will allow the counsil to decide which projects will be continously funded. Each tuple represent the address and the respecting amount of NAM that will receive every epoch. The list of funding recipients will be stored in storage under the PGF internal address substorage space.
+
+```rust
+struct cPgfRecipients {
```

Comment:
Probably we should have a transaction to add a recipient and a transaction to remove a recipient, just for ease of use by the council

Reply:
bump

Reply:
I think I added this